src/test/java/com/aws/greengrass/mqttbridge/MQTTBridgeTest.java

```diff
@@ -344,4 +337,29 @@ private void updateConfig(String[] config, String value) throws ServiceLoadExcep
         kernel.locate(MQTTBridge.SERVICE_NAME).getConfig().lookupTopics(configPath)
                 .updateFromMap(Utils.immutableMap(configName, value), updateBehavior);
     }
+
+    private CountDownLatch trackServiceStateChanges(State... expectedStates) {
+        final Queue<State> states = new LinkedList<>(Arrays.asList(expectedStates));
+        CountDownLatch stateChangesOccurred = new CountDownLatch(1);
+        kernel.getContext().addGlobalStateChangeListener((GreengrassService service, State was, State newState) -> {
+            if (!service.getName().equals(MQTTBridge.SERVICE_NAME)) {
+                return;
+            }
+            synchronized (states) {
+                if (states.isEmpty()) {
+                    return;
```

Comment:
Not sure when this can occur unless method is called with an empty list. If it does, we should countdown the latch.

Reply:
other way it could occur is if another state transition occurs after all the ones we care about have been reached.  say `trackServiceStateChanges(NEW)` and we reinstall a service and let the test run through startup, then this would also trigger `RUNNING` after `NEW`, which would hit this condition